### PR TITLE
Check /etc/os-release for the host's ident if /etc/issue contains only \S

### PR DESCRIPTION
**Pull request by zsck**

### CODE REVIEW

File: mig-agent/agentcontext/env_linux.go [hunk 1]
```diff
@@ -75,11 +75,21 @@ func findOSInfo(orig_ctx AgentContext) (ctx AgentContext, err error) {
...
 	}
 	logChan <- mig.Log{Desc: fmt.Sprintf("getIssue() failed: %v", err)}.Debug()
+
+	ctx.OSIdent, err = getOSRelease()
+	if err == nil {
+		logChan <- mig.Log{Desc: "using /etc/os-release for distribution ident"}.Debug()
+		goto haveident
+	}
+
 	logChan <- mig.Log{Desc: "warning, no valid linux os identification could be found"}.Info()
```
Comment (ameihm0912) [inline mig-agent/agentcontext/env_linux.go]: Should this just return at this point?

File: mig-agent/agentcontext/env_linux.go [hunk 2]
```diff
@@ -75,11 +75,21 @@ func findOSInfo(orig_ctx AgentContext) (ctx AgentContext, err error) {
...
+
+	// Here we check that we read more than '\S'.
+	// See https://access.redhat.com/solutions/1138953
+	if err == nil && len(ctx.OSIdent) > 3 {
 		logChan <- mig.Log{Desc: "using /etc/issue for distribution ident"}.Debug()
 		goto haveident
 	}
 	logChan <- mig.Log{Desc: fmt.Sprintf("getIssue() failed: %v", err)}.Debug()
+
+	ctx.OSIdent, err = getOSRelease()
```
Comment (ameihm0912) [inline mig-agent/agentcontext/env_linux.go]: It doesn't look like we log the error response here like we do with getIssue()

Comment (zsck) [reply mig-agent/agentcontext/env_linux.go]: Right you are! This is fixed.

File: mig-agent/agentcontext/env_linux.go (hunk 1 above)
Comment (zsck) [reply mig-agent/agentcontext/env_linux.go]: This function also populates the `AgentContext`'s `Init` field a couple lines down from here.

**Comment by zsck** [comment]
@ameihm0912 Going to request a re-review when you have time.  Any idea what could be causing the builds in Travis to fail?``

**Comment by zsck** [comment]
Errors seem to be changing all the time.

I'm not sure why there's a problem importing the `osext` module.  I've been able to run all the mig tests locally just fine with

```
go clean --testcache
make test
```

**Pull request merged**